data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,32 @@
+local function levitate(creature, parameter)
+	local fromPosition = creature:getPosition()
+
+	if parameter == "up" and fromPosition.z ~= 8 or parameter == "down" and fromPosition.z ~= 7 then
```

Comment:
You can store the direction on a variable to avoid having to query `parameter == "up"` all the time. Not sure how fast Lua is with strings, but it is definitely faster with booleans.